static/src/javascripts/lib/geolocation.js

```diff
@@ -0,0 +1,133 @@
+// @flow
+import fetch from 'lib/fetch-json';
+import config from 'lib/config';
+import storage from 'lib/storage';
+
+let location;
+const storageKey = 'gu.geolocation';
+const editionToGeolocationMap = {
+    UK: 'GB',
+    US: 'US',
+    AU: 'AU',
+};
+const daysBeforeGeolocationRefresh = 10;
+
+const get = (): Promise<string> =>
+    new Promise((resolve, reject) => {
+        if (location) return resolve(location);
+
+        fetch('/geolocation', {
+            method: 'GET',
+            contentType: 'application/json',
+            mode: 'cors',
+        })
```

Comment:
If we have to keep the `Content-Type` header, this has to be written as:

```
fetchJSON('/geolocation', {
    headers: {
        'Content-Type': 'application/json',
    }
    mode: 'cors',
})
```

Could you also rename `fetch` to `fetchJSON` to make it clear, we aren't using fetch here?